airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -1057,6 +1071,196 @@ def _delete_cluster(self, hook: DataprocHook):
+    ) -> None:
+        super().__init__(**kwargs)
+        self.project_id = project_id
+        self.region = region
+        self.cluster_name = cluster_name
+        self.cluster_uuid = cluster_uuid
+        self.request_id = request_id
+        self.retry = retry
+        self.timeout = timeout
+        self.metadata = metadata
+        self.gcp_conn_id = gcp_conn_id
+        self.impersonation_chain = impersonation_chain
+
+    def _get_cluster(self, hook: DataprocHook) -> Cluster:
+        """
+        Retrieve the cluster information.
+
+        :param hook: Required. Instance of ``airflow.providers.google.cloud.hooks.dataproc.DataprocHook`` class
+            to interact with Dataproc API
+        :return: Instance of ``google.cloud.dataproc_v1.Cluster``` class
+        """
+        return hook.get_cluster(
+            project_id=self.project_id,
+            region=self.region,
+            cluster_name=self.cluster_name,
+            retry=self.retry,
+            timeout=self.timeout,
+            metadata=self.metadata,
+        )
+
+    def _check_desired_cluster_state(self, hook: Cluster) -> (bool, str | None):
+        """Implement this method in child class to return whether the cluster is in desired state or not.
+
+        If the cluster is in desired stated you can return a log message content as a second value
+        for the return tuple.
+
+        :param hook: Required. Instance of ``airflow.providers.google.cloud.hooks.dataproc.DataprocHook``
+            class to interact with Dataproc API
+        :return: Tuple of (Boolean, Optional[str]) The first value of the tuple is whether the cluster is
+            in desired state or not. The second value of the tuple will use if you want to log something when
+            the cluster is in desired state already.
+        """
+
+        return NotImplemented
+
+    def _get_operation(self, hook: DataprocHook) -> operation.Operation:
+        """Implement this method in child class to call the related hook method and return its result.
+
+        :param hook: Required. Instance of ``airflow.providers.google.cloud.hooks.dataproc.DataprocHook``
+            class to interact with Dataproc API
+        :return: ``google.api_core.operation.Operation`` value whether the cluster is in desired state or not
+        """
+        return NotImplemented
+
+    def _get_return_value(self, hook: DataprocHook) -> Any | None:
+        """This method can be overridden if you want to return something from the execute method.
+
+        This method's return value will be also return from the execute function.
+        This method returns None as default, if you want to return something else, then you can override it.
+        """
+
+        return None
+
+    def execute(self, context: Context):
+        hook = DataprocHook(gcp_conn_id=self.gcp_conn_id, impersonation_chain=self.impersonation_chain)
+        cluster: Cluster = self._get_cluster(hook)
+        print("CLUSTER", cluster)
```

Comment:
Could you change this row from `print()` to `self.log.info` or remove if this row was added for test purpose?

Reply:
it was just for test purposes, I will remove the line.